opensrp-chw/src/main/java/org/smartregister/chw/interactor/DefaultPncUpcomingServiceInteractorFlv.java

```diff
@@ -41,71 +48,118 @@
+                    serviceName = context.getString(R.string.up_immunizations, context.getString(R.string.at_birth), baby.getFirstName(), context.getString(R.string.bcg));
+                }
+            }
         }
+
+        return serviceName;
+
+
     }
 
     private void evaluateImmunization(List<BaseUpcomingService> serviceList) {
         Date deliveryDate = new Date();
         Date OverDueDate = new Date();
-        Map<String, String> alerts = ChwPNCDao.getPNCImmunizationAtBirth(memberObject.getBaseEntityId());
         BaseUpcomingService upcomingService = new BaseUpcomingService();
+        int count = 0;
         try {
             deliveryDate = new SimpleDateFormat("dd-MM-yyyy").parse(PncLibrary.getInstance().profileRepository().getDeliveryDate(memberObject.getBaseEntityId()));
             OverDueDate = new DateTime(deliveryDate).plusDays(13).toDate();
         } catch (Exception e) {
+            Timber.v(e.toString());
         }
-        if (Days.daysBetween(new DateTime(new DateTime(deliveryDate).plusDays(27).toDate()).toLocalDate(), new DateTime().toLocalDate()).getDays() < 28) {
-            if (alerts.size() > 0) {
-                for (Map.Entry<String, String> alert : alerts.entrySet()) {
-                    if (alert.getValue().equalsIgnoreCase("Vaccine not given")) {
-                        count += count;
-                    }
-                }
-            }
-            if (alerts.size() == 0 || count == 2) {
-                upcomingService.setServiceName(context.getString(R.string.upcoming_immunizations, context.getString(R.string.at_birth), context.getString(R.string.bcg), context.getString(R.string.opv_0)));
-            } else {
-                for (Map.Entry<String, String> alert : alerts.entrySet()) {
-                    if (alert.getKey().equalsIgnoreCase(context.getString(R.string.opv_0))) {
-                        upcomingService.setServiceName(context.getString(R.string.up_immunizations, context.getString(R.string.at_birth), context.getString(R.string.opv_0)));
-                    } else {
-                        upcomingService.setServiceName(context.getString(R.string.up_immunizations, context.getString(R.string.at_birth), context.getString(R.string.bcg)));
-                    }
-                }
+        List<PncBaby> pncBabies = PersonDao.getMothersPNCBabies(memberObject.getBaseEntityId());
+        for (PncBaby baby : pncBabies) {
+            if (immunizationServiceName(baby) != null) {
+                upcomingService.setServiceName(immunizationServiceName(baby));
+                upcomingService.setServiceDate(deliveryDate);
+                upcomingService.setOverDueDate(OverDueDate);
+                count += 1;
             }
-            upcomingService.setServiceDate(deliveryDate);
-            upcomingService.setOverDueDate(OverDueDate);
+        }
+        if (count > 0) {
```

Comment:
Not sure what the logic here should be doing, but it appears we're looping through all the babies but in effect only adding the last baby's upcoming service to the list. The rest of the other babies' services won't be added. 
I'd suggest the ` serviceList.add(upcomingService);` be moved inside the ` for (PncBaby baby : pncBabies) {
                for (Map.Entry<String, String> alert : alerts.entrySet()) {	            if (immunizationServiceName(baby) != null) {` block